### PR TITLE
fix: continues #115, song disk bug

### PR DESCRIPTION
**Pull request by thamelodev**
It's simple, I've provided the fix to this bug in Arcturus already, but not in React yet, took a long time, because I was working with another projects.

But to explain, it's simple, it gets the song name from the offset 5 in stuffData that has the format of a LegacyDataType, this method ``getLegacyStringWithOffset`` was not in nitro-renderer, for that reason I've opened some PR there also: billsonnn/nitro-renderer#105, If the song name wasn't found in stuffName, just set it to the old ``SONG_NAME``

inventory

Fixed in Arcturus Community (ms4/dev), found that was related to a packet [Merge Request in Arcturus](https://git.krews.org/morningstar/Arcturus-Community/-/merge_requests/16)

1avatarinfo

TLDR: This will fix the song name in inventory, showing the correct name

Fixes: #86 
Continues: #115

### CODE REVIEW

**Comment by dank074** [comment]
The SWF doesn't seem to handle it this way.

snippet from the swf:
image

We can't expect all emulators to adhere to the stuff data format that you added to Morningstar.

**Comment by duckietm** [comment]
But instead of adding the stuff to the renderer, is this not a better option:

`case FurniCategory.TRAX_SONG:
        const songInfo =
          GetNitroInstance().soundManager.musicController.getSongInfo(
            k.extra
          ) as SongDataEntry;
        if (songInfo == null) break;
        // remove the event if we already know the song
        this.clearTraxSongInfoEvent();
        this._name = songInfo.creator;
        return;`

**Comment by duckietm** [comment]
is this not a better solutions, no renderer update required:

private setName(): void {
    const k = this.getLastItem();

    if (!k) {
      this._name = "";

      return;
    }

    let key = "";

    switch (this._category) {
      case FurniCategory.POSTER:
        key = "poster_" + k.stuffData.getLegacyString() + "_name";
        break;
      case FurniCategory.TRAX_SONG:
        const songInfo =
          GetNitroInstance().soundManager.musicController.getSongInfo(
            k.extra
          ) as SongDataEntry;
        if (songInfo == null) break;
        // remove the event if we already know the song
        this.clearTraxSongInfoEvent();
        this._name = songInfo.creator;
        return;
      case FurniCategory.FLOOR:
        key = "inventory.furni.item.floor.name";
        break;
      case FurniCategory.WALL_PAPER:
        key = "inventory.furni.item.wallpaper.name";
        break;
      case FurniCategory.LANDSCAPE:
        key = "inventory.furni.item.landscape.name";
        break;
      default:
        if (this.isWallItem) {
          key = "wallItem.name." + k.type;
        } else {
          key = "roomItem.name." + k.type;
        }
    }

    this._name = LocalizeText(key);
  }
